Allow override of `-couch_ini` parameter

adrienverge opened the pull request:
This is the continuation of commit 649b808 "Allow override of
`-args_file` and `-config` parameters (#1095)".

This allows setting the `-couch_ini` files location if needed, using
`COUCHDB_INI_FILES` env var:
- If unset (the default), `-couch_ini` argument won't be passed.
- If set (to a file, or list of files like
  `COUCHDB_INI_FILES=/chroot/etc/couchdb/default.ini`
  `/chroot/etc/couchdb/default.d ...`) it will allow running CouchDB with
  different settings. A useful case is being able to run multiple
  CouchDB servers on the same machine.

adrienverge commented on rel/files/couchdb.in:
```diff
@@ -31,8 +31,10 @@ export COUCHDB_QUERY_SERVER_COFFEESCRIPT="{{prefix}}/bin/couchjs {{prefix}}/shar
 
 exec "$BINDIR/erlexec" -boot "$ROOTDIR/releases/$APP_VSN/couchdb" \
      -args_file "${ARGS_FILE}" \
+     -couch_ini ${INI_FILES} \
```
No `"` here, because `$INI_FILES` can reference zero or more files, and they must be handled as such.

adrienverge commented:
Hello, what do you think of this change?

Our team have been patching CouchDB with this (for years now), to be able to run multiple
CouchDB servers on the same machine.

adrienverge commented:
Hello team, what you think of this simple (yet useful) change?

wohali commented:
@adrienverge Are you sure this works correctly when specifying no list? My read of the source code suggests that it won't:

https://github.com/apache/couchdb-config/blob/master/src/config_app.erl#L30-L48

and:

```
$ erl -a b
Erlang/OTP 21 [erts-10.2.4] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:1]

Eshell V10.2.4  (abort with ^G)
1> init:get_argument(a).
{ok,[["b"]]}
2>
User switch command
 --> q
$ erl -a
Erlang/OTP 21 [erts-10.2.4] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:1]

Eshell V10.2.4  (abort with ^G)
1> init:get_argument(a).
{ok,[[]]}
2>
User switch command
 --> q
$ erl
Erlang/OTP 21 [erts-10.2.4] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:1]

Eshell V10.2.4  (abort with ^G)
1> init:get_argument(a).
error
2>
User switch command
 --> q
```

Because an empty list returns success, I think this overrides the default order. Can you confirm?

adrienverge commented:
@wohali thanks for your answer, and valuable feedback.

I've updated the pull request to even avoid passing `-couch_ini` if the env var is unset, so that the command-line args really stay the same in the default case. I've also updated the commit message accordingly.
```diff
@@ -31,10 +31,10 @@ export COUCHDB_QUERY_SERVER_COFFEESCRIPT="{{prefix}}/bin/couchjs {{prefix}}/shar
 export COUCHDB_FAUXTON_DOCROOT={{fauxton_root}}
 
 ARGS_FILE="${COUCHDB_ARGS_FILE:-$ROOTDIR/etc/vm.args}"
-INI_FILES=${COUCHDB_INI_FILES:-}
+[ -n "${COUCHDB_INI_FILES:-}" ] && INI_ARGS="-couch_ini $COUCHDB_INI_FILES"
 SYSCONFIG_FILE="${COUCHDB_SYSCONFIG_FILE:-$ROOTDIR/releases/$APP_VSN/sys.config}"
 
 exec "$BINDIR/erlexec" -boot "$ROOTDIR/releases/$APP_VSN/couchdb" \
      -args_file "${ARGS_FILE}" \
-     -couch_ini ${INI_FILES} \
+     ${INI_ARGS:-} \
      -config "${SYSCONFIG_FILE}" "$@"
```

Do you think it's OK?

wohali commented:
+1. We need passing tests (ugh) and I'm happy to merge this as the final step in making all of those CLI parameters overrideable.

adrienverge commented:
> We need passing tests (ugh)

Sure. This is like the 12th time I push-force to have them run again; I'll keep doing it until they pass.

Thanks!

adrienverge commented:
Hi, all tests pass now :)
